### PR TITLE
[RLlib] Raise err on deprecated

### PR DESCRIPTION
**Pull request by kouroshHakha**
## Why are these changes needed?

Old deprecations should raise errors messages.

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [x] This PR is not tested :(

### CODE REVIEW

**Comment by gjoliver** [comment]
we haven't been very diligent in actually deleting deprecated stuff.
are we trying to do this by 1.13, so we can delete everything from 2.0?

**Comment by sven1977** [comment]
@gjoliver , that's the plan.
1.13: Turn warnings into errors
2.0: Fully deprecated (no warnings or errors).

File: rllib/agents/ppo/ppo.py [hunk 1]
```diff
@@ -239,7 +239,7 @@ def __init__(self):
     @Deprecated(
         old="ray.rllib.agents.ppo.ppo.DEFAULT_CONFIG",
         new="ray.rllib.agents.ppo.ppo.PPOConfig(...)",
-        error=False,
+        error=True,
```
Comment (ArturNiederfahrenhorst) [inline rllib/agents/ppo/ppo.py:242]: Are we moving to PPOConfig **now**? What about other configs? Feels kind of weird hard decprecating the PPO config on 1.13.

Comment (ArturNiederfahrenhorst) [reply rllib/agents/ppo/ppo.py:242]: I guess it's just too tedious to keep both for now, eh?

**Review by ArturNiederfahrenhorst — APPROVED** [review]
Thanks @kouroshHakha 🥇

File: rllib/agents/ppo/ppo.py (hunk 1 above)
Comment (sven1977) [reply rllib/agents/ppo/ppo.py:242]: Oh yeah, good catch!

File: rllib/utils/sgd.py [hunk 2]
```diff
@@ -48,21 +48,27 @@ def minibatches(samples: SampleBatch, sgd_minibatch_size: int, shuffle: bool = T
...
-        for i, j in data_slices:
-            yield samples[i:j]
-    else:
-        all_slices = list(zip(data_slices, state_slices))
-        if shuffle:
-            # Make sure to shuffle data and states while linked together.
-            random.shuffle(all_slices)
-        for (i, j), (si, sj) in all_slices:
-            yield samples.slice(i, j, si, sj)
+    #all_slices = samples._get_slice_indices(sgd_minibatch_size)
```
Comment (kouroshHakha) [inline rllib/utils/sgd.py:51]: Why are these changes part of this PR?

**Pull request closed without merging**